zilliqa/tests/it/zil.rs

```diff
@@ -2257,9 +2269,78 @@ async fn get_smart_contract_sub_state(mut network: Network) {
     assert!(substate2.get("welcome_msg").is_none());
 }
 
-#[allow(dead_code)]
-async fn get_soft_confirmed_transaction(mut _network: Network) {
-    todo!();
+#[zilliqa_macros::test]
+async fn get_soft_confirmed_transaction(mut network: Network) {
+    let wallet = network.random_wallet().await;
+
+    // Create a Zilliqa account and get its secret key and address
+    let (secret_key, _address) = zilliqa_account(&mut network).await;
+
+    // Define the recipient address
+    let address_string_w_prefix = "0x00000000000000000000000000000000deadbeef";
+    let to_addr: H160 = address_string_w_prefix.parse().unwrap();
+
+    // Send a transaction
+    let (_contract_address, returned_transaction) = send_transaction(
+        &mut network,
+        &secret_key,
+        1,
+        ToAddr::Address(to_addr),
+        200u128 * 10u128.pow(12),
+        50_000,
+        None,
+        None,
+    )
+    .await;
+
+    // Get the transaction ID from the returned transaction
+    let transaction_id = returned_transaction["ID"]
+        .as_str()
+        .expect("Failed to get ID from response");
+
+    // Wait for the transaction to be mined
+    network.run_until_block(&wallet, 1.into(), 50).await;
+
+    // Use the GetTransaction API to retrieve the transaction details
+    let response: Value = wallet
+        .provider()
+        .request("GetTransaction", [transaction_id])
```

Comment:
Isn't this the wrong API? Thought FWIW I'd probably prefer just adding an extra request to the existing `GetTransaction` test which asserts that the response from `GetTransaction` and `GetSoftConfirmedTransaction` are the same.

Reply:
Right, I've combined them, and good spot on the copy paste error